cpp/CMakePresets.json

```diff
@@ -368,6 +421,16 @@
       "displayName": "Release build with Gandiva",
       "cacheVariables": {}
     },
+    {
+      "name": "ninja-release-python-emscripten",
+      "inherits": [
+        "features-emscripten",
+        "base-release",
+        "features-python"
```

Comment:
it is in as
```
"features-emscripten",
"base-release"
```
and 
```
"features-emscripten",
"base-debug"
```
It needs to be in that order because things from base need overriding (base sets 'ARROW_BUILD_STATIC' off.)